android/src/main/java/com/reactnativecommunity/asyncstorage/AsyncLocalStorageUtil.java

```diff
@@ -142,4 +146,36 @@ private static void deepMergeInto(JSONObject oldJSON, JSONObject newJSON)
       }
     }
   }
+  /**
+   * From Pie and up, Android started to use Write-ahead logging (WAL), instead of journal rollback
+   * for atomic commits and rollbacks.
+   * Basically, WAL does not write directly to the database file, rather to the supporting WAL file.
+   * Because of that, migration to the next storage might not be successful, because the content of
+   * RKStorage might be still in WAL file instead. Committing all data from WAL to db file is called
+   * a "checkpoint" and is done automatically (by default) when the WAL file reaches a threshold
+   * size of 1000 pages.
+   * More here: https://sqlite.org/wal.html
+   * <p>
+   * This helper will force checkpoint on RKStorage, if Next storage file does not exists yet.
+   */
+  public static void verifyAndForceSqliteCheckpoint(Context ctx) {
+    File nextStorageFile = ctx.getDatabasePath("AsyncStorage");
+    File currentStorageFile = ctx.getDatabasePath(ReactDatabaseSupplier.DATABASE_NAME);
+    boolean isCheckpointRequired = Build.VERSION.SDK_INT >= Build.VERSION_CODES.P
+            && !nextStorageFile.exists()
+            && currentStorageFile.exists();
+    if (!isCheckpointRequired) {
+      Log.i("AsyncStorage_Next", "SQLite checkpoint not required.");
+      return;
+    }
+
+    try {
+      ReactDatabaseSupplier supplier = ReactDatabaseSupplier.getInstance(ctx);
+      supplier.get().execSQL("PRAGMA wal_checkpoint");
+      supplier.closeDatabase();
+      Log.i("AsyncStorage_Next", "Forcing SQLite checkpoint successful.");
+    } catch (Exception e) {
+      Log.w("AsyncStorage_Next", "Could not force checkpoint on RKStorage: " + e.getMessage());
```

Comment:
Should we also add that we cannot guarantee that the migrated db may not contain the latest?